index.ts

```diff
@@ -46,3 +46,31 @@ export function createPatch(input, output): Operation[] {
   // a new Pointer gets a default path of [''] if not specified
   return diffAny(input, output, ptr);
 }
+
+/**
+Produce an 'application/json-patch+json'-type patch to test a patch before
+working with existing values.
+
+This does not alter `input` or `output` unless they have a property getter with
+side-effects (which is not a good idea anyway).
+
+Returns list of test operations to perform on `input` to validate `patch`.
+*/
+export function createTests(input, patch: Operation[]): Operation[] {
+  let tests = new Array<Operation>();
```

Comment:
I generally go with `const` wherever possible. Perhaps there are performance optimizations that prefer let, though?